pseudopotential_calculator/calculations/adsorbate.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal, Self, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+from ase import Atom, Atoms
+from matplotlib import cm
+
+from pseudopotential_calculator.calculations.generic import (
+    OptimizationParamsBase,
+    repeat_cell,
+)
+from pseudopotential_calculator.calculations.slab import get_n_th_slab_layer_height
+from pseudopotential_calculator.castep import (
+    CastepConfig,
+    get_calculator_atom,
+    get_default_calculator,
+)
+
+if TYPE_CHECKING:
+    from ase.calculators.castep import Castep
+    from matplotlib.figure import Figure
+
+
+class AdsorbateOptimizationParams(OptimizationParamsBase):
+    """Parameters of a adsorbate calculation."""
+
+    @property
+    def kpoint_mp_grid(self: Self) -> str:
+        return f"{self.n_k_points} {self.n_k_points} {1}"
+
+
+def get_adsorbate_optimization_calculator(
```

Comment:
maybe add a get_calculator() in generic which does everything which is shared between these three. Maybe everything is shared (and there is actually no longer a benefit to AdsorbateOptimizationParams)